File: Go/Solutions/projeuler/Problem020.go
package euler

import (
	"fmt"
	"time"
)

/*
Problem020 answers the problem at : https://projecteuler.net/problem=20

* Problem 20:
		Find the sum of the digits in the number 100!
*/
func Problem020() {
	start := time.Now()
	ans := DigitalRoot(Factorial(100))
	end := time.Now()

	fmt.Printf("\nAnswer to Problem 20 : %d\n", ans)
	fmt.Printf("Time Taken: %f seconds\n\n", end.Sub(start).Seconds())
}

File: Python/Problem020.py
from math import factorial
from time import time

''' 
Problem 20: https://projecteuler.net/problem=20

Find the sum of the digits in the number 100!
'''

digits = lambda n: map(int, list(str(n)))

start = time()
ans = sum(digits(factorial(100)))

if __name__ == '__main__':
    print(f"\nAnswer: { ans }")
    print(f"Time Taken: { time() - start }\n")
